### PR TITLE
Evidence session filtering/ flagging optimal and suboptimal responses

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
add color coding for suboptimal response counts for each conjunction in each session in the session index view; add "Reached Optimal" field in individual session view for each prompt

## WHY
this will make it easier for Curriculum to parse session data

## HOW
add `because_optimal`, `but_optimal` and `so_optimal` values to query used for session index view; add function to color code response counts and add logic to calculate "Reached Optimal" with existing `attempts` data

### Screenshots
(If applicable. Also, please censor any sensitive data)
Screen Shot 2022-11-02 at 2 20 57 PM
Screen Shot 2022-11-02 at 2 21 08 PM

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/Activity-Session-Filtering-Flag-Responses-that-Are-Optimal-Sub-Optimal-ce25c0407a6d41be99c6e9880470a3a5

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | yes

### CODE REVIEW

File: services/QuillLMS/app/models/feedback_history.rb [hunk 1]
```diff
@@ -251,8 +251,11 @@ def self.list_by_activity_session(activity_id: nil, page: 1, start_date: nil, en
         comprehension_prompts.activity_id,
         ARRAY_REMOVE(ARRAY_AGG(DISTINCT feedback_history_flags.flag), NULL) AS flags,
         COUNT(CASE WHEN comprehension_prompts.conjunction = '#{BECAUSE}' THEN 1 END) AS because_attempts,
+        BOOL_OR(CASE WHEN comprehension_prompts.conjunction = '#{BECAUSE}' AND feedback_histories.optimal = true THEN true ELSE false END) AS because_optimal,
```
Comment (brendanshean) [inline services/QuillLMS/app/models/feedback_history.rb:282]: TIL`BOOL_OR`.  Cool

**Review by brendanshean — APPROVED** [review]
Nice job on figuring out that query.

File: services/QuillLMS/client/app/bundles/Staff/components/evidence/activitySessions/promptTable.tsx [hunk 2]
```diff
@@ -37,12 +39,15 @@ const PromptTable = ({ activity, rules, prompt, showHeader, sessionId }: PromptT
         });
       });
       completed = usedAttempts.some((attempt) => attempt.optimal);
+      optimal = completed ? true : false
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/evidence/activitySessions/promptTable.tsx]: Doesn't `.some` return a boolean? If it does this line should be unnecessary (you're just setting optimal to the value of completed)

**Review by emilia-friedberg — APPROVED** [review]

File: services/QuillLMS/client/app/bundles/Staff/components/evidence/activitySessions/promptTable.tsx (hunk 2 above)
Comment (eadams17) [reply services/QuillLMS/client/app/bundles/Staff/components/evidence/activitySessions/promptTable.tsx]: oh good catch, I'll update

**Pull request merged**